### PR TITLE
Unstack

### LINKED ISSUE #11543: unstack

### PR DESCRIPTION
**Pull request by RheagalFire**
Close #11543

### CODE REVIEW

**Comment by RheagalFire** [comment]
I am getting this error while running the tests, the ground truth returned a list of two arrays.
image

File: ivy/functional/frontends/tensorflow/general_functions.py [hunk 1]
```diff
@@ -347,21 +347,23 @@ def roll(input, shift, axis, name=None):
...
+    name=None,
 ):
     return ivy.repeat(input, repeats, axis=axis)
+
+
+@to_ivy_arrays_and_back
+@with_unsupported_dtypes({"1.11.0 and below": ("float16", "bfloat16")}, "torch")
+def unstack(value: ivy.array, axis=0, num=None, name=None):
+    res = ivy.unstack(value, axis=axis)
+    return ivy.astype(res, value.dtype)
```
Comment (ZiadAmerr) [inline ivy/functional/frontends/tensorflow/general_functions.py:369]: You should remove `ivy.astype(res, value.dtype)`

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_general_functions.py [hunk 2]
```diff
@@ -1566,5 +1565,32 @@ def test_tensorflow_repeat(
...
+@handle_frontend_test(
+    fn_tree="tensorflow.unstack",
+    dtypes_value_axis=helpers.dtype_values_axis(
+        available_dtypes=helpers.get_dtypes("float"),
+        num_arrays=1,
+        min_value=-100,
+        max_value=100,
+        min_num_dims=2,
+        # shape=st.shared(helpers.get_shape(min_num_dims=2), key="shape"),
+    ),
```
Comment (ZiadAmerr) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_general_functions.py:1581]: You can replace all these except `available_dtypes` with `min_num_dims` and `min_dim_size`.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_general_functions.py [hunk 3]
```diff
@@ -1566,5 +1565,32 @@ def test_tensorflow_repeat(
...
+    x_dtype, x, axis = dtypes_value_axis
+    print("Following are the variables: ", x_dtype, x[0], axis)
+    helpers.test_frontend_function(
+        input_dtypes=x_dtype,
+        frontend=frontend,
+        test_flags=test_flags,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        value=x[0]
+        # axis=axis,
```
Comment (ZiadAmerr) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_general_functions.py:1595]: You should also add the `axis` to consideration

File: ivy/functional/frontends/tensorflow/general_functions.py (hunk 1 above)
Comment (RheagalFire) [reply ivy/functional/frontends/tensorflow/general_functions.py:369]: If I remove that I get an error for mismatching data type from the ground truth any fixes to solve that?

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_general_functions.py (hunk 2 above)
Comment (RheagalFire) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_general_functions.py:1581]: How can I ensure that the range of the axis remains in the range of [-R, R), R being the rank of tensor to be unstacked?

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_general_functions.py (hunk 3 above)
Comment (RheagalFire) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_general_functions.py:1595]: How can I ensure that the range of the axis remains in the range of [-R, R), R being the rank of tensor to be unstacked?

Comment (RheagalFire) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_general_functions.py:1595]: Ok solved this just need to add num=None as an argument in the testing function.

Comment (RheagalFire) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_general_functions.py:1595]: I want to pass axis as a keyword argument but it is taking it as a positional argument.

Comment (RheagalFire) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_general_functions.py]: Ok, seems like the axis was getting passed as a positional argument and therefore it was being interpreted as `num` argument. Fixed this by adding `number_positional_args=st.just(1)`

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_general_functions.py (hunk 2 above)
Comment (RheagalFire) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_general_functions.py:1579]: Done

File: ivy/functional/frontends/tensorflow/general_functions.py (hunk 1 above)
Comment (RheagalFire) [reply ivy/functional/frontends/tensorflow/general_functions.py]: Removed.

**Comment by RheagalFire** [comment]
I am running the test locally for my function and it is passing all the test cases. @WilliamHirst Can you assist?

**Comment by RheagalFire** [comment]
@ZiadAmerr All tests are getting passed now. Can you review my changes?

**Comment by ZiadAmerr** [comment]
I can't review my changes I am just voluntarily helping you, the person responsible for reviewing the changes is @WilliamHirst, he will review it as soon as possible :)

**Comment by RheagalFire** [comment]
Seems like there was a force push that has caused some problem, in local, the test cases are not passing for the JAX backend 
image
Once this is resolved , i will modify my PR.

**Pull request merged**